src/components/TheTableOfContents.vue

```diff
@@ -0,0 +1,253 @@
+<template>
+  <AppFlex
+    direction="col"
+    gap="none"
+    hAlign="stretch"
+    class="toc"
+    :style="{ top: nudge + 'px' }"
+    :data-expanded="expanded"
+    role="doc-toc"
+    aria-label="Page table of contents"
+    @click.stop
+  >
+    <!-- toggle button -->
+    <AppFlex gap="none" hAlign="left" class="title">
+      <button
+        class="button"
+        @click="expanded = !expanded"
+        :aria-expanded="expanded"
+        v-tippy="
+          expanded ? 'Close table of contents' : 'Expand table of contents'
+        "
+      >
+        <AppIcon :icon="expanded ? 'times' : 'bars'" />
+      </button>
+      <span v-if="expanded" class="title-text">Table of Contents</span>
+    </AppFlex>
+
+    <template v-if="expanded">
+      <div class="spacer"></div>
+
+      <!-- entries -->
+      <AppLink
+        v-for="(entry, index) in entries"
+        :key="index"
+        :to="'#' + entry.id"
+        class="entry"
+        :data-active="active === index"
+        @click="active = index"
+        :aria-current="active === index"
+      >
+        <AppIcon :icon="entry.icon" class="entry-icon" />
+        <span class="entry-text truncate">{{ entry.text }}</span>
+      </AppLink>
+
+      <div class="spacer"></div>
+
+      <!-- options -->
+      <AppCheckbox
+        v-model="oneAtATime"
+        text="Solo section"
```

Comment:
Super minor point, but "Solo section" took me a bit to parse (although the tooltip helps a lot). Perhaps something along the lines of "Show single section" or "Only one section" would be more parseable...?

Reply:
Sure, I'll update this text. "Soloing" something is a term that comes from music production, so I have bias here.

Reply:
Also, I tried putting just the whole tooltip text as the visible text, but it was too long and wrapped and looked ugly, so just trying to come up with a more terse phrase.

I changed it to "show single section".